packages/ui/react-ui-list/src/components/Tree/TreeItem.tsx

```diff
@@ -54,7 +54,7 @@ export type TreeItemProps<T = any> = {
   path: string[];
   last: boolean;
   draggable?: boolean;
-  renderColumns?: FC<{ item: T; path: string[]; menuOpen: boolean; setMenuOpen: (open: boolean) => void }>;
+  renderItem?: FC<{ item: T; path: string[]; menuOpen: boolean; setMenuOpen: (open: boolean) => void }>;
```

Comment:
why render item? imo render columns makes more sense because this allows a consumer to control the rendering of the additional grid columns for each item. The rendering of the item itself is not overridable and always controlled by the tree. Perhaps a jsdoc would be helpful here.

Reply:
the tree has no concept of columns. i found this really confusing since this function renders a row!

Reply:
it's a treegrid, it has columns https://github.com/dxos/dxos/blob/main/packages/ui/react-ui-list/src/components/Tree/Tree.tsx#L23

I can see how it could be interpreted as rendering the whole column as a single component, but it renders the column one row at a time. It's explicitly not the whole item though, only the additional column content for each row

Reply:
i still think this is a leaky abstraction since tree doesn't know anything about columns per se. i.e., gridTemplateColumns should just be getRowClassName?

Reply:
it's explicitly a tree grid, so not so sure about that

Reply:
but let me take another look since i do see what you mean; i need to understand this better (which was the point of the exercise)